### ISSUE TITLE
Cursor moves from UPS_CURSOR_FIRST to UPS_CURSOR_PREVIOUS without UPS_KEY_NOT_FOUND

### ISSUE THREAD

**Issue by billon-pl** [billon-pl/upscaledb#4]
This is issue is identical to issue-107  from https://github.com/cruppstahl/upscaledb 
See https://github.com/cruppstahl/upscaledb/issues/107

Please consider sample:

``` c++
int main()
{
    ups_env_t* env;
    ups_env_create(&env, "test.db", UPS_ENABLE_TRANSACTIONS, 0664, 0);
    //ups_env_create(&env, "test.db", 0, 0664, 0);

    ups_parameter_t params[] = {
    {UPS_PARAM_KEY_TYPE, UPS_TYPE_UINT32},
    {0, }
    };

    ups_db_t* db;
    ups_env_create_db(env, &db, 1, 0, &params[0]);

    const int item_count = 57; // required

    for (int i = 0; i < item_count; i++)
    {
        ups_key_t key = ups_make_key(&i, sizeof(i));
        ups_record_t record = {0};

        ups_db_insert(db, 0, &key, &record, 0);
    }

    ups_cursor_t* cur;

    ups_cursor_create(&cur, db, 0, 0);

    int key_data = 0;
    ups_key_t key = {0}; key.data = &key_data; key.size = sizeof(key_data);
    ups_record_t rec = {0};

    ups_cursor_move(cur, &key, &rec, UPS_CURSOR_FIRST);
    if((UPS_SUCCESS == ups_cursor_move(cur, &key, &rec, UPS_CURSOR_PREVIOUS)))
    {
        std::cout << "Cursor moved from FIRST to PREVIOUS without UPS_KEY_NOT_FOUND" << std::endl;
    }

    ups_cursor_close(cur);

    ups_db_close(db, 0);

    return 0;
}
```
In standard output:
```
    Cursor moved from FIRST to PREVIOUS without UPS_KEY_NOT_FOUND
```
Issue occurs for `item_count=57` and not for example for `item_count=50`
Transactions must be enabled to observe the issue

**Comment by billon-pl** [comment]
- This case is added to test suite of the project. 

- The source code can be found in file [issue107.cc](https://github.com/billon-pl/upscaledb/tree/master/unittests/issue107.cc)

- There is work around for this issue: use function `ups_cursor_find` without using function `ups_cursor_move`.

**Comment by billon-pl** [comment]
- The answer to this issue could be found in the comment of the source code of function [LocalDb::cursor_move](https://github.com/billon-pl/upscaledb/blob/28af73f187fd870a4d6a11ec37234e8e7f156174/src/4db/db_local.cc#L1357). 

- This comment explains that cursors (in some situations) could be circular. 

- When the cursor is circular one, then calling the function `ups_cursor_move(cur, &key, &rec, UPS_CURSOR_PREVIOUS)` returns success even when the cursor is at the "beginning" of the cursor. 

- In other words, **there is no begin and no end in the circular cursors**.
